src/light.jl

```diff
@@ -165,13 +166,15 @@ source(::LightAutomaton, t::LightTransition) = t.edge.src
 event(A::LightAutomaton, t::LightTransition) = A.Σ[t.edge][t.id]
 target(::LightAutomaton, t::LightTransition) = t.edge.dst
 
-function in_transitions(A::LightAutomaton, s)
-    edges = MappedArrays.mappedarray(src -> edge_object(A, src, s),
+function in_transitions(A::LightAutomaton{GT, ET}, s) where {GT, ET}
+    edges = MappedArrays.mappedarray(ET,
```

Comment:
I misinterpreted the first argument, sorry. If you give 3 arguments to `mappedarray`, the first one is `f`, the second is the inverse of `f` (used in `setindex!`) and the third one is the array to map.
When the first argument is a type, then it is used as the return type and it does not try a testvalue of zero to try to guess the type (which is what we want).
However, here the function we want to apply is not a type so it does not work. I suggest we just bypass the `mappedarray` function and directly do
```julia
f(src) = edge_object(A, src, s)
inneigh = LightGraphs.inneighbors(A.G, s)
edges = MappedArrays.ReadonlyMappedArray{ET, 1, typeof(f), typeof(inneigh)}(f, inneigh)
```

Reply:
Thanks, I'll try.

Reply:
Should it not be the other way around (swap type parameters 3 and 4)? See [here](https://github.com/JuliaArrays/MappedArrays.jl/blob/a8189572df9ca2db3d28fd74d1684c24c1d56c26/src/MappedArrays.jl#L10).

Reply:
Yes indeed!